docs/components.md

```diff
@@ -101,127 +83,54 @@ To learn more about lifecycle methods, [see the lifecycle methods page](lifecycl
-
-```javascript
-function ClosureComponent(initialVnode) {
-	// Each instance of this component has its own instance of `kind`
-	var kind = "closure component"
-
-	return {
-		view: function(vnode) {
-			return m("div", "Hello from a " + kind)
-		},
-		oncreate: function(vnode) {
-			console.log("We've created a " + kind)
-		}
-	}
-}
-```
-
-The returned object must hold a `view` function, used to get the tree to render.
-
-They can be consumed in the same way regular components can.
-
-```javascript
-// EXAMPLE: via m.render
-m.render(document.body, m(ClosureComponent))
-
-// EXAMPLE: via m.mount
-m.mount(document.body, ClosureComponent)
-
-// EXAMPLE: via m.route
-m.route(document.body, "/", {
-	"/": ClosureComponent
-})
-
-// EXAMPLE: component composition
-function AnotherClosureComponent() {
-	return {
-		view: function() {
-			return m("main",
-				m(ClosureComponent)
-			)
-		}
-	}
-}
-```
-
-If a component does *not* have state then you should opt for the simpler POJO component to avoid the additional overhead and boilerplate of the closure.
-
-#### ES6 classes
+### Passing data to components
 
-Components can also be written using ES6 class syntax:
+Data can be passed to component instances by passing an `attrs` object as the second parameter in the hyperscript function:
 
 ```javascript
-class ES6ClassComponent {
-	constructor(vnode) {
-		this.kind = "ES6 class"
-	}
-	view() {
-		return m("div", `Hello from an ${this.kind}`)
-	}
-	oncreate() {
-		console.log(`A ${this.kind} component was created`)
-	}
-}
+m(Example, {name: "Floyd"})
 ```
 
-Component classes must define a `view()` method, detected via `.prototype.view`, to get the tree to render.
-
-They can be consumed in the same way regular components can.
+This data can be accessed in the component's view or lifecycle methods via the `vnode.attrs`:
 
 ```javascript
-// EXAMPLE: via m.render
-m.render(document.body, m(ES6ClassComponent))
-
-// EXAMPLE: via m.mount
-m.mount(document.body, ES6ClassComponent)
-
-// EXAMPLE: via m.route
-m.route(document.body, "/", {
-	"/": ES6ClassComponent
-})
-
-// EXAMPLE: component composition
-class AnotherES6ClassComponent {
-	view() {
-		return m("main", [
-			m(ES6ClassComponent)
-		])
+var Example = {
+	view: function (vnode) {
+		return m("div", "Hello, " + vnode.attrs.name)
 	}
 }
 ```
 
-#### Mixing component kinds
-
-Components can be freely mixed. A class component can have closure or POJO components as children, etc...
+NOTE: Lifecycle methods can also be provided via the `attrs` object, so you should avoid using the lifecycle method names for your own callbacks as they would also be invoked by Mithril. Use lifecycle methods in `attrs` only when you specifically wish to create lifecycle hooks.
```

Comment:
This would flow a little better and sound less repetitive if you used a few more pronouns. I've also suggested a few other tweaks here.

```suggestion
NOTE: Lifecycle methods can also be defined in the `attrs` object, so you should avoid using their names for your own callbacks as they would also be invoked by Mithril itself. Use them in `attrs` only when you specifically wish to use them as lifecycle methods.
```

Reply:
Sure, that's just what was there. Yours is better!